gemstone/generator/generator.py

```diff
@@ -43,23 +65,39 @@ def add_ports(self, **kwargs):
         for name, T in kwargs.items():
             self.add_port(name, T)
 
+    def __hash_wire(self, connection):
+        self.__hash ^= (~(hash(connection[0]) ^ hash(connection[1]))) << 16
+
+    def set_hash(self, new_hash_value):
+        self.__hash = new_hash_value
+
+    def __hash__(self):
+        return self.__hash
+
     def wire(self, port0, port1):
-        assert isinstance(port0, PortReferenceBase)
```

Comment:
same re: debug mode?

Reply:
There are millions of `isinstance` calls from gemstone. This change reduces a significant portion of it.